pandas/core/generic.py

```diff
@@ -9276,16 +9276,11 @@ def rank(
             msg = "na_option must be one of 'keep', 'top', or 'bottom'"
             raise ValueError(msg)
 
-        def ranker(data):
-            if data.ndim == 2:
-                # i.e. DataFrame, we cast to ndarray
-                values = data.values
-            else:
-                # i.e. Series, can dispatch to EA
-                values = data._values
-
-            if isinstance(values, ExtensionArray):
-                ranks = values._rank(
+        def ranker(blk_values):
+            if axis_int == 0:
+                blk_values = blk_values.T
```

Comment:
i think we can avoid transposing at the block level and only transpose at the dataframe level (L9317).  then pass axis=blk_values.ndim-1 (which will require updating the EA method)